frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1028,3 +1032,22 @@ func autoDetectPlatform(img Image, target specs.Platform, supported []specs.Plat
 	}
 	return target
 }
+
+func getPlatformArgs(bp, tp specs.Platform) []instructions.KeyValuePairOptional {
+	m := map[string]string{
+		"BUILDPLATFORM":  platforms.Format(bp),
```

Comment:
Maybe we should add underscore prefixes to these reserved args?

Reply:
cc @thaJeztah

Reply:
> If the user happens to use or wants to use an ARG with the same name they can do that and it overrides the automatic values.

They can but they should not?
Maybe we should print warning, because people may use the same name accidentally without being aware of this PR

Reply:
Actually someone is already using `ENV TARGETOS=.. TARGETARCH=..` in his/her own Dockerfile

https://raw.githubusercontent.com/ARwMq9b6/dnsproxy/master/cmd/dnsproxy/Dockerfile

Reply:
Users should always get the behavior they expect. If these vars are used in the Dockerfile and user doesn't know about the automatic values and uses them as regular ones they will behave as regular ones. Only way the automatic behavior applies is if the Dockerfile author knows about the automatic behavior and uses these vars without passing in the values.

Reply:
For the link, there is no behavior change. This Dockerfile will continue to work exactly like before.

To fill these vars automatically they would need to expose it as arg in a stage and not provide the value(eg. explicitly break the current behavior)

Reply:
Namespacing these will be tricky yes; not sure it underscores would work (esthetically, it feels ugly, but of course that's personal).

we're a bit limited because arg double as env vars, so (officially) dots aren't allowed, which could've been used for namespacing (then again, we don't want these names to become too long)

Separate from the namespacing discussion; wondering if they should be named `TARGET_OS` and `TARGET_ARCH`

Reply:
For `TARGET_OS`, the current is similar to `GOOS/GOARCH`

Reply:
hm yeah 🤔